sound/soc/sof/ipc4-topology.c

```diff
@@ -207,6 +207,14 @@ static int sof_ipc4_get_audio_fmt(struct snd_soc_component *scomp,
 	if (!base_config)
 		return -ENOMEM;
 
+	available_fmt->input_audio_fmts = kcalloc(available_fmt->audio_fmt_num,
+						  sizeof(struct sof_ipc4_audio_format),
+						  GFP_KERNEL);
+	if (!available_fmt->input_audio_fmts) {
+		kfree(base_config);
+		return -ENOMEM;
+	}
+
```

Comment:
is there a reason why we do a loop line 218..227 for a single object. Can't we use a single call to sof_update_ipc_object() as fone line 230 for ibs/obs?